src/components/TodoFooter.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import { Todo } from '../types/Todo';
+import { Filter } from '../types/Filter';
+import classNames from 'classnames';
+
+type Props = {
+  todos: Todo[];
+  filter: Filter;
+  setFilter: (filter: Filter) => void;
+  handleDeleteCompleted: () => void;
+};
+
+export const TodoFooter: React.FC<Props> = ({
+  todos,
+  filter,
+  setFilter,
+  handleDeleteCompleted,
+}) => {
+  const todosLeft = todos.filter(todo => !todo.completed).length;
+  const hasCompleted = todos.some(todo => todo.completed);
+
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="TodosCounter">
+        {`${todosLeft} items left`}
+      </span>
+
+      <nav className="filter" data-cy="Filter">
+        <a
+          href="#/"
+          className={classNames('filter__link', {
+            selected: filter === Filter.All,
+          })}
+          data-cy="FilterLinkAll"
+          onClick={() => setFilter(Filter.All)}
+        >
+          All
+        </a>
+
+        <a
+          href="#/active"
+          className={classNames('filter__link', {
+            selected: filter === Filter.Active,
+          })}
+          data-cy="FilterLinkActive"
+          onClick={() => setFilter(Filter.Active)}
+        >
+          Active
+        </a>
+
+        <a
+          href="#/completed"
+          className={classNames('filter__link', {
+            selected: filter === Filter.Completed,
+          })}
+          data-cy="FilterLinkCompleted"
+          onClick={() => setFilter(Filter.Completed)}
+        >
+          Completed
+        </a>
```

Comment:
Consider using `Object.values(Filter)` and `map()` method

Reply:
should I use it instead of words 'All', 'Active', and ' Completed'?

Reply:
> should I use it instead of words 'All', 'Active', and ' Completed'?

instead of three links you can use something like that:
```
{Object.values(Filter).map(filterItem => (
   <a
          href={`#/${filterItem}`}
          className={classNames('filter__link', {
            selected: filter === filterItem,
          })}
          onClick={() => setFilter(filterItem)}
        >
          {filterItem}
        </a>
))}
```

Reply:
Still not fixed from the previous review, you need to fix it and after that create a helper function for add data-cy attributes and call it here